client/call.go

```diff
@@ -29,6 +30,7 @@ type Call struct {
 	Args  proto.Request  // The argument to the command
 	Reply proto.Response // The reply from the command
 	Err   error          // Error during call creation
+	post  func() error   // Function to be called after successful completion
```

Comment:
This seems useful for general-purpose unmarshaling; we should probably export it so apps that use something other than protos can provide their own hook here. (Would it be worthwhile to call the hook even if there is an error? I.e. are there possible uses for this hook other than unmarshaling?)

It's a little odd that this hook requires the use of a closure; should the hook take the Reply (or the entire Call) as an argument? OTOH, I don't see how you avoid the use of a closure for the thing that you're unmarshaling into, so maybe it's simpler to just use the closure for everything.